contrib/platform/test/com/sun/jna/platform/win32/Kernel32Test.java

```diff
@@ -332,14 +332,20 @@ public void testOpenProcess() {
     }
 
     public void testQueryFullProcessImageName() {
-        HANDLE h = Kernel32.INSTANCE.OpenProcess(0, false, Kernel32.INSTANCE.GetCurrentProcessId());
-        assertNotNull("Failed (" + Kernel32.INSTANCE.GetLastError() + ") to get process handle", h);
-    	
-        char[] path = new char[WinDef.MAX_PATH];
-        IntByReference lpdwSize = new IntByReference(path.length);
-        boolean b = Kernel32.INSTANCE.QueryFullProcessImageName(h, 0, path, lpdwSize);
-        assertTrue("Failed (" + Kernel32.INSTANCE.GetLastError() + ") to query process image name", b);
-        assertTrue("Failed to query process image name, empty path returned", lpdwSize.getValue() > 0);
+        HANDLE h = null;
+        try {
+	        h = Kernel32.INSTANCE.OpenProcess(0, false, Kernel32.INSTANCE.GetCurrentProcessId());
+	        assertNotNull("Failed (" + Kernel32.INSTANCE.GetLastError() + ") to get process handle", h);
+	    	
+	        char[] path = new char[WinDef.MAX_PATH];
+	        IntByReference lpdwSize = new IntByReference(path.length);
+	        boolean b = Kernel32.INSTANCE.QueryFullProcessImageName(h, 0, path, lpdwSize);
+	        assertTrue("Failed (" + Kernel32.INSTANCE.GetLastError() + ") to query process image name", b);
+	        assertTrue("Failed to query process image name, empty path returned", lpdwSize.getValue() > 0);
+        } finally {
+        	if (h != null)
+        		Kernel32.INSTANCE.CloseHandle(h);
+        }
     }
```

Comment:
You don't need to initialize the handle and then check whether it is null before closing it. The _assertNotNull_ will take care of it:
```java
HANDLE h = ...obtain process..
assertNotNull("Failed ...", h);
try {
   ...run the tests...
} finally {
    CloseHandle(h); // the handle is implicitly not NULL because of the assertion...
}